core/domain/state_domain.py

```diff
@@ -173,6 +174,170 @@ def validate(self):
+
+    def validate(self):
+        """Validates all properties of an Image.
+
+        Raises:
+            ValidationError: One or more attributes of the ImageAssets are not
+            valid.
+        """
+        if not isinstance(self.placeholder, bool):
+            raise utils.ValidationError(
+                'Expected placeholder to be bool, received %s' %
+                self.placeholder)
+        if not isinstance(self.instructions, basestring):
+            raise utils.ValidationError(
+                'Expected instructions to be string,' +
+                'received %s' %self.instructions)
+
+
+class ImageAssets(object):
+    """Value object representing the images for the state."""
+    def __init__(self, image_mapping):
+        """Constructs a ImageAssets domain object.
+
+        Args:
+            image_mapping: dict. The dict representation of
+                ImageAssets mapping.
+        """
+        self.image_mapping = image_mapping
+        self.validate()
+
+    def validate(self):
+        """Validates all properties of ImageAssets.
+
+        Raises:
+            ValidationError: One or more attributes of the ImageAssets are not
+            valid.
+        """
+        for image_id in self.image_mapping:
+            image_re = r'(image_id_)+[0-9]'
+            if not re.match(image_re, image_id):
+                raise utils.ValidationError(
+                    'Invalid image_id')
+            image = self.image_mapping[image_id]
+            image.validate()
+
+    def to_dict(self):
+        """Returns a dict representing this ImageAssets domain object.
+
+        Returns:
+            dict. A dict, mapping all fields of ImageAssets instance.
+        """
+        image_assets = {}
+        image_mapping = {}
+
+        for image_id in self.image_mapping:
+            image = self.image_mapping[image_id]
+            image_mapping[image_id] = image.to_dict()
+
+        image_assets['image_mapping'] = image_mapping
+        return image_assets
+
+    @classmethod
+    def from_dict(cls, image_assets_dict):
+        """Return a ImageAssets domain object from a dict.
+
+        Args:
+            image_assets_dict: dict. The dict representation of
+                ImageAssets object.
+
+        Returns:
+            ImageAssets. The corresponding ImageAssets domain
+            object.
+        """
+        image_mapping = {}
+        for image_id in image_assets_dict['image_mapping'].iterkeys():
+            image_info = image_assets_dict['image_mapping'][image_id]
+            image_mapping[image_id] = Image.from_dict(image_info)
+
+        return cls(image_mapping)
+
+    def add_image(self, image_id, image_info):
+        """Adds default image object in state.
+
+        Args:
+            image_id: int. The image_id of an image.
+            image_info: dict. The dicts representation of image info.
+        """
+        image_re = r'(image_id_)+[0-9]'
+        if not re.match(image_re, image_id):
```

Comment:
Do we need `re` here? I think we can do this validation directly right? Also, if you are going to use this check in multiple places than create a sperate method to validate image_id.

Reply:
If we are expecting image_id inside ImageAssets to be in a specific pattern then this method should go inside ImageAssets class.